python/ray/serve/_private/proxy_state.py

```diff
@@ -30,35 +32,174 @@
+    def start_new_drained_check(self):
+        """Start a new drained check on the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_ready(self) -> Any:
+        """Return the payload from proxy ready check when ready."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_healthy(self) -> bool:
+        """Return whether the proxy actor is healthy or not."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_drained(self) -> bool:
+        """Return whether the proxy actor is drained or not."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def check_health(self):
+        """Check the health of the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def update_draining(self, draining: bool):
+        """Update the draining status of the proxy actor."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def kill_proxy(self):
+        """Kill the proxy actor."""
+        raise NotImplementedError
+
+
+class ProxyActorWrapper(ActorWrapper):
+    def start_new_ready_check(self):
+        """Start a new ready check on the proxy actor."""
+        self._ready_obj_ref = self._actor_handle.ready.remote()
+
+    def start_new_health_check(self):
+        """Start a new health check on the proxy actor."""
+        self._health_check_obj_ref = self._actor_handle.check_health.remote()
+
+    def start_new_drained_check(self):
+        """Start a new drained check on the proxy actor."""
+        self._is_drained_obj_ref = self._actor_handle.is_drained.remote(
+            _after=self._update_draining_obj_ref
+        )
+
+    def is_ready(self) -> Any:
+        """Return the payload from proxy ready check when ready."""
+        finished, _ = ray.wait([self._ready_obj_ref], timeout=0)
+        if finished:
+            return ray.get(finished[0])
+
+        return None
+
+    def is_healthy(self) -> bool:
+        """Return whether the proxy actor is healthy or not."""
+        finished, _ = ray.wait([self._health_check_obj_ref], timeout=0)
+        if finished:
+            self._health_check_obj_ref = None
+            return ray.get(finished[0])
+
+        return False
+
+    def is_drained(self) -> bool:
+        """Return whether the proxy actor is drained or not."""
+        finished, _ = ray.wait([self._is_drained_obj_ref], timeout=0)
+        if finished:
+            self._is_drained_obj_ref = None
+            return ray.get(finished[0])
+
+        return False
+
+    def check_health(self):
+        """Check the health of the proxy actor."""
+        ray.get(self._actor_handle.check_health.remote(), timeout=0.001)
+
+    def update_draining(self, draining: bool):
+        """Update the draining status of the proxy actor."""
+        self._update_draining_obj_ref = self._actor_handle.update_draining.remote(
+            draining, _after=self._update_draining_obj_ref
+        )
+
+    def kill_proxy(self):
```

Comment:
```suggestion
    def kill(self):
```
none of the other methods have `_proxy` suffix